subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ComponentMetadataRulesIntegrationTest.groovy

```diff
@@ -241,7 +287,7 @@ dependencies {
             }
 
             resolve.doLast {
-                assert rulesInvoked.sort() == [ 1, 2, 3 ]
+                assert rulesInvoked.sort() == [ 1, 2, 3, 4 ]
```

Comment:
:x: Same comment, this looks wrong. We should probably check what the component metadata suppliers do, because it's likely they have the same problem. Probably we could move the responsibility to isolate properly to the instantiating action.